### PR TITLE
Add some more options to the flake8 action

### PR DESCRIPTION
**Pull request by peternewman**
This also includes #8 .

### CODE REVIEW

**Comment by peternewman** [comment]
Hi @TrueBrain ,

Is this good to merge please? This does enough for me to run this against my codebase and I'd rather use the "official" one if possible.

File: entrypoint.sh [hunk 1]
```diff
@@ -9,9 +9,19 @@ echo "::add-matcher::${RUNNER_TEMP}/_github_workflow/flake8-matcher.json"
 # The weird piping is needed because we want to get the exitcode from flake8,
 # but there is no other good universal way of doing so.
 # e.g. PIPESTATUS and pipestatus only work in bash/zsh respectively.
-echo "Running flake8 on '${INPUT_PATH}' ..."
+echo "Running flake8 on '${INPUT_PATH}' with the following options..."
+command_args=""
+echo "Ignoring '${INPUT_IGNORE}'"
+if [ "x${INPUT_IGNORE}" != "x" ]; then
```
Comment (TrueBrain) [inline entrypoint.sh:15]: I always prefer: `[ -z "${INPUT_IGNORE}" ]` over the "x" method. But this is more preference than anything else :D

**Comment by TrueBrain** [comment]
Owh, and pretty please document these two options in the `README.md`, so other also know about it :) That would be awesome!

**Comment by peternewman** [comment]
> Owh, and pretty please document these two options in the `README.md`, so other also know about it :) That would be awesome!

:facepalm: Done. Could do with a way to auto-generate documentation from action.yml files.

File: entrypoint.sh (hunk 1 above)
Comment (peternewman) [reply entrypoint.sh:15]: I thought the x thing is slightly more cross-compatible or safer if the variable doesn't exist or something (although I guess quoting may catch that). Happy to change it if you'd prefer.

**Comment by peternewman** [comment]
> I am so sorry; for some reason I did not get a notification about this! Happy you gave a poke :) And tnx for this PR, this is awesome :D

Don't worry, I wasn't sure if you were busy given the current situation.

> One request: would you mind adding a few test cases for this?
> 
> So in https://github.com/TrueBrain/actions-flake8/blob/master/example/example.py, make a line of like 90 and 110 chars, set the max length to 100 in https://github.com/TrueBrain/actions-flake8/blob/master/.github/workflows/testing.yml. That should trigger only one. And make an error/warning you put on ignore, so we can see that doesn't happen.
> 
> Makes reviewing PRs easier, as that means I can spot if the output is as expected :D

I can go one better and nick the good work of @per1234 in https://github.com/codespell-project/actions-codespell/pull/7 and give you some proper tests and CI if you'd like. Or can it wait for another PR?

Here's a live example with those options turned on FWIW:
https://github.com/peternewman/ola/actions/runs/123856886

**Comment by TrueBrain** [comment]
Having a proper CI would be awesome, but indeed something for another PR :)

Btw, often it is better to add a `.flake8` file in the repo to set ignores and line length, but I don't mind adding this at all :) more options is better :D

**Comment by peternewman** [comment]
> Btw, often it is better to add a `.flake8` file in the repo to set ignores and line length, but I don't mind adding this at all :) more options is better :D

Thanks, that's good to know and might be easier keeping our Travis CI and GitHub action run config in sync, I guess it depends how much we want to clutter the repo with config files

**Pull request merged**